pageserver/src/basebackup_cache.rs

```diff
@@ -173,6 +224,11 @@ impl BasebackupCache {
 
         encoder.shutdown().await?;
 
+        self.data_cache
+            .lock()
+            .await
+            .insert((tenant_shard_id.tenant_id, timeline_id), lsn);
```

Comment:
What if it already exists?

In that case we will forget that we have another entry and will never remove it.

I guess we may want to store an array as an value,

or, as an alternative, if we have `tokio::sync::mpsc::unbounded_channel` for deleting entries (and drain it in `select!`), then we may insert a new entry into the `data_cache` and add previous LSN (if existed) to `deletion_queue`, so the background task will delete it

The second option is probably simpler (don't need to keep an array)